stablehlo/dialect/VhloDialect.h

```diff
@@ -0,0 +1,54 @@
+/* Copyright 2023 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_DIALECT_VHLO_DIALECT_H
+#define STABLEHLO_DIALECT_VHLO_DIALECT_H
+
+#include "llvm/ADT/StringRef.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/Dialect.h"
+#include "mlir/IR/DialectImplementation.h"
+
+namespace mlir {
+namespace vhlo {
+
+class VhloDialect : public Dialect {
```

Comment:
Ah now I see what you meant when you asked offline what I think about creating VhloDialect.h.

What seems unfortunate about the proposed arrangement is that we don't have VhloDialect.cpp (which we cannot have because we'll be back to circular dependencies), so depending on the `vhlo_dialect` target alone is insufficient for the program to link correctly. Let's try to avoid that.

It looks like the only dependency from `VhloTypes.h` to `VhloDialect.h` is `VhloDialect::getCurrentVersion()`. Given that, how do you feel about moving `getCurrentVersion` (and `getMinimumVersion`) into `VhloVersion.h` instead of creating this file and this target?

This is not perfect from the purity perspective (I remember we discussed this some time ago), but it seems preferable to having the `vhlo_dialect` target as it stands right now.